### PR TITLE
Bug fix reset password via email 182201722

### PR DESCRIPTION
**Pull request by HiHeritier**
**What does this PR do?**

feature to reset password

**Description of Task to be completed?**

- User can be able to reset password

**How should this be manually tested?**

- Clone the repo checkout to branch `bug-fix-reset-password-via-email-182201722`
- Run `npm install` to install modules
- Run `npm run start:prod` to start the production server
- In your API testing tool (postman recommended ) call api `/api/v1/user/forgotpassword` to have the token to reset password sent to your email
-Copy the token sent to your email then call api `/api/v1/user/resetpassword/$token`  to inter new password

**What are the relevant pivotal tracker stories?**
https://www.pivotaltracker.com/story/show/182201722

**Screenshots**
Screenshot (196)

Screenshot (197)

Screenshot (198)

### CODE REVIEW

**Review by Placide123 — COMMENTED** [review]
on my side, it provides a reset message but the token provided is not helping to reset the password (it is providing `Token is invalid or has expired ), others can test. if it works the problem will be at my end; if not you will make some changes to let it works.

**Review by semy-tuyi — COMMENTED** [review]
Screen Shot 2022-07-08 at 16 39 45
try to figure out why swagger is giving that error.
I am getting error "invalid time stamp" when I try to `POST` `api/v1/user/forgotpassowrd`
Screen Shot 2022-07-08 at 16 39 22

**Review by Placide123 — DISMISSED** [review]

**Review by uwambajeddy — DISMISSED** [review]

**Review by semy-tuyi — CHANGES_REQUESTED** [review]
fix this on your swagger documentation please
Screen Shot 2022-07-15 at 14 51 57
Testing this feature locally I am getting this error.
Screen Shot 2022-07-15 at 14 54 13

**Review by uwambajeddy — CHANGES_REQUESTED** [review]
can't find reset password end points on swagger

**Review by Patricknshimiyimana — CHANGES_REQUESTED** [review]
While working on the issues above, you may also rebase on develop and resolve conflicts

**Review by Placide123 — APPROVED** [review]

**Review by semy-tuyi — APPROVED** [review]

**Review by alaintresor — APPROVED** [review]

**Review by uwambajeddy — APPROVED** [review]
LGTM

**Review by Patricknshimiyimana — APPROVED** [review]

**Review by mwisemarierose — APPROVED** [review]

File: src/routers/api/authentication.js [hunk 1]
```diff
@@ -1,32 +1,7 @@
 import express from 'express';
-import {
```
Comment (joshwambere) [inline src/routers/api/authentication.js:2]: why did you delete those lines

File: src/routers/api/authentication.js [hunk 2]
```diff
@@ -1,32 +1,7 @@
...
-  signup,
-  verifyEmail,
-} from '../../controllers/authentication';
-import passport, { facebookCallBack } from '../../config/passport';
-import { authenticate } from 'passport';
+import { signup } from '../../controllers/authentication';
 const authRouter = express.Router();
 
 authRouter.post('/signup', signup);
-authRouter.get('/logout', protect, logout);
```
Comment (joshwambere) [inline src/routers/api/authentication.js:15]: and here

File: src/routers/api/users.js [hunk 3]
```diff
@@ -33,21 +35,6 @@ const uploads = multer({ storage, fileFilter });
 const userRouter = express.Router();
 
 userRouter.post('/signup', signup);
-userRouter.post('/logout', logout);
```
Comment (joshwambere) [inline src/routers/api/users.js:36]: why did you delete those lines two

File: test/authentication.test.js [hunk 4]
```diff
@@ -170,49 +167,3 @@ describe('User login', () => {
       });
   });
 });
-describe('User logout', () => {
```
Comment (joshwambere) [inline test/authentication.test.js:173]: and those tests?

File: test/tripRequest.test.js [hunk 5]
```diff
@@ -587,7 +586,6 @@ describe('Reject Trip Request', () => {
       .end((err, res) => {
         const { message } = res.body;
         expect(res.status).to.equal(401);
-        expect(message).to.equal('Your token is invalid or expired');
```
Comment (joshwambere) [inline test/tripRequest.test.js:590]: and why did you delete this line also?

**Review by joshwambere — CHANGES_REQUESTED** [review]

**Pull request closed without merging**